libwasmvm/src/api.rs

```diff
@@ -254,8 +261,60 @@ impl BackendApi for GoApi {
             Err(e) => return (Err(BackendError::unknown(e.to_string())), gas_info),
         };
         callee_instance.env.set_serialized_env(&contract_env);
-        callee_instance.set_storage_readonly(caller_env.is_storage_readonly());
         gas_info.cost += instantiate_cost;
+        // set read-write permission to callee instance
+        let callee_info = FunctionMetadata {
+            module_name: String::from(&func_info.module_name),
+            name: "_list_callable_points".to_string(),
+            signature: ([], [Type::I32]).into(),
+        };
+        let callee_ret = match callee_instance.call_function_strict(
```

Comment:
Before call `_list_callable_points`, the instance needs to be readonly. This is because, if `_list_callable_points` is defined as read/write function (as we not expected) it can change the storage.